### PR TITLE
proposed architecture section to documentation

### PR DESCRIPTION
**Pull request by sparrell**
adds in component, monolithic, and multi architectures

### CODE REVIEW

File: docs/Arch/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+Another reason might be different vendors/open-source-projects.
+P1-P14 are fourteen different products (e.g. from a vendor)
+or projects (e.g. built from open source).
+
+Yet another reason might be geographical.
+E.g. One PCS for each of five regions.
+
+In Figure 3 above, there are five PCS, four PAR, three PES and
+two monolithic PACE systems,
+each with there own interface to the integration bus.
```
Comment (dlemire60) [inline docs/Arch/README.md]: typo  "there" --> "their"

**Comment by dlemire60** [comment]
In the use case "Putting an SBOM into PACE", the collected attributes are shown as going from component -to-> PCS -to-> PAR, which seems inefficient. I think a better path would be for the collection function on the component to transfer the requested data to the PAR directly and notify the PCS when finished. Then the data is only transferred once per component queried instead of twice. So the sequence of actions would be adjusted from the current proposal:

1. Decision-making element (e.g a SOAR running a CACAO playbook eg {fillinhere}) sends OpenC2 command, (eg {fillinhere}), telling PCS to retrieve SBOM from device_id=2345.
2. PCS receives command and sends OpenC2 commands {fillinhere} to device, 
3. Device retrieves SBOM as directed and stores info in PAR via PAR API
4. Device sends OpenC2 response to PCS indicating command result
6. PCS notifies Decision-Making Element command completed successfully {fillinhere}

If the command in {1} identified multiple devices for SBOM retrieval then the PCS would run {2,3,4} in parallel for all devices before sending the notice of completion in {5}.

I'm also envisioning that the PCS in OpenC2 terms is a combined Consumer / Producer, and we'll need to start developing a PCS Actuator Profile that will define the command(s) to be used in {1}.

**Comment by dlemire60** [comment]
My comment above is equivalent to one I made to an OpenC2 SBOM Actuator Profile issue:
https://github.com/oasis-tcs/openc2-ap-sbom/issues/34

which was accepted in principle during last week's OpenC2 working meeting (2/23).

**Comment by sparrell** [comment]
Although I agreed with @dlemire60 comments during OpenC2 meeting, I am no longer as confident it is doable as proposed and I am not as sure anymore that is simpler. My issue is with directionality of the commands and with permissions issues. I don't think it's a good idea to have the command come from the PCS and the response go to the PAR. So to implement the proposal would mean the PCS tells the device to send a command to the PAR directly. I don't think devices should be on the command side of the interface. I'm worried about the security model and who keeps track of who is allowed to do what. I think we'd end up building much of the PCS functionality into both the PAR and into the devices themselves. But maybe I'm just missing something.

**Comment by dlemire60** [comment]
pcs_01-DPL

Here's my re-draw of @sparrell's image envisioning direct delivery of the SBOM from component to PAR; I was envisioning a simple HTTP PUT for the delivery; perhaps the permissions issue could be handled by passing an API key or similar from the PCS to the Component to grant permission to write to the PAR.

**Comment by davaya** [comment]
I support merging this PR.
After reconsideration, I agree that the component-write-to-PAR option raises security concerns.

Based on the SCAPv2 work, one message flow (using one of the Component, Monolithic, or Multi PACE options) would be for the decision-making (DM) component to query the PAR first.  The PAR would provide stored attribute and/or evaluation responses along with their timestamps.  If the data is sufficiently complete and fresh, the DM is done.  If not, the DM could request the PCS / PES to update the PAR with more/newer data, then retrieve results from the PAR when the PCS/PES is done.

This allows the PAR external interface to be read-only, cache-friendly, and cheap to invoke.  And actual work would only be done when necessary.  I think this allows Asset Inventory (#28) to be an implementation detail inside the PACE boundary, since it doesn't have a published interface.  DMs discover assets through the PAR interface, but filling the PAR is internal implementation.

**Comment by dlemire60** [comment]
I'm amenable to the concerns that @sparrell and @davaya have raised regarding the Component writing directly to the PAR. I am now a bit confused WRT to the division of roles among the DM / PCS / PES in @davaya's description. If the DM queries the PAR and uses the results to make it's decision, doesn't that mean the DM is usurping the role of the PES (which we haven't discussed at all yet in PACE)? At what point is the PES doing any "evaluation"? I guess if viewed in RFC 7632 terms, the PAR might be multiple repositories, with the PES writing evaluation results to one of them and that's where the information provided to the DM actually originates. That would be consistent with the depiction I created in the attached PDF.

[PACE_Process_Per_RFC7632.pdf](https://github.com/opencybersecurityalliance/PACE/files/8170206/PACE_Process_Per_RFC7632.pdf)

EDIT: Delivery of attributes from the PCS to the PAR is also consistent with the _ad hoc_ collection scenario described in the [SACM Architecture ID](https://www.ietf.org/archive/id/draft-ietf-sacm-arch-13.txt); see section 6.4 and subs, pp 34-39.

**Comment by davaya** [comment]
The PAR interface (Duncan's "blue wire in the three-pin connector" for Monolithic) would be able to return results from all repositories, whether implemented as a single or multiple storage services.  In RFC 7632 terms the Posture Attribute Values and Posture Evaluation Results would both be accessible through the PAR interface and would have a cache lifetime for fresh results, after which they would become stale but retained for a forensic lifetime.  (Freshness and forensic lifetimes could be set to zero, of course.)

**Pull request merged**